class.jetpack-autoupdate.php

```diff
@@ -42,18 +41,21 @@ private function __construct() {
 			add_action( 'shutdown', array( $this, 'log_results' ) );
 		}
 
-		if ( is_main_site() ) {
-			// Anytime WordPress saves update data, we'll want to update our Jetpack option as well.
-			add_action( 'set_site_transient_update_plugins', array( $this, 'save_update_data' ) );
-			add_action( 'set_site_transient_update_themes', array( $this, 'save_update_data' ) );
-			add_action( 'set_site_transient_update_core', array( $this, 'save_update_data' ) );
+		$jetpack = Jetpack::init();
+		// TODO: needs capability check before syncing updates, but current_user is not available
+		$jetpack->sync->mock_option( 'updates', array( $this, 'get_updates' ) );
+		$jetpack->sync->mock_option( 'update_details', array( $this, 'get_update_details' ) );
```

Comment:
you are right, now I get it, this sync thing uses all the WP advanced stuff combined :scream: